python/cudf/cudf/io/json.py

```diff
@@ -37,16 +37,31 @@ def read_json(
             f"or a bool, or None. Got {type(dtype)}"
         )
 
-    if engine == "cudf" and not lines:
+    if engine == "cudf_experimental":
+        raise ValueError(
+            "engine='cudf_experimental' support has been removed, "
+            "use `engine='cudf'`"
+        )
```

Comment:
Should this be a hard error or a deprecation warning that replaces the value with `"cudf"`?

Reply:
Since it was experimental, I feel we have the flexibility to make this an error, what do you think?

Reply:
I think that's justifiable -- but we'll want to delete this error at some point later. Therefore, it's the same amount of work for us as developers to deprecate it as to force an error. "Add warning, delete warning later" vs. "add error, delete error later."